routes/messages/commands/get_other_categories.js

```diff
@@ -0,0 +1,35 @@
+const UserCategories = require('models/db/userCategories.js');
+const Categories = require('models/db/categories.js');
+const QuickReplyTemplate = require('../UI/QuickReplyTemplate.js');
+
+module.exports = async event => {
+    const { user_id } = event;
+
+    const userCategories = await UserCategories.retrieve({ user_id });
+    const userCategoryIDs = userCategories.map(cat => {
+        return cat.category_id;
+    });
+
+    const allCategories = await Categories.retrieve();
+
+    const otherCategories = [];
+
+    allCategories.forEach(cat => {
+        if (!userCategoryIDs.includes(cat.id)) {
+            otherCategories.push(cat);
+        } 
+    });
```

Comment:
It won't make much of a difference in the actual runtime because we don't have very many categories.  However, if you want to be super efficient, it's constant time to look a key up in an object but O(n) to find an entry in an array.

So instead of a `.map` you could do:
```
const userCategoryIDs = {};
for (let c in userCategories) {
  userCategoryIDs[c.category_id] = true;
}

allCategories.forEach(cat => {
  if (!userCategoryIDs[c.id]) {
    otherCategories.push(cat);
  }
});
```

But again, because we have a really small number of categories, shouldn't make any noticeable difference.

If you want to keep the code clean too, you could also use a Map for the lookup structure:

```
const userCategoryIDs = new Map(
  userCategories.map(({ category_id }) => [category_id, true])
);

allCategories.forEach(cat => {
  if (!userCategoryIDs.get(c.id)) {
    otherCategories.push(cat);
  }
});
```

Same runtime as the object, just a little cleaner to write.